include/seqan/basic/iterator_interface.h

```diff
@@ -350,43 +350,12 @@ position(TIterator const & it,
 
 template <typename T, typename TContainer>
 inline SEQAN_HOST_DEVICE bool
-atBegin(T const & it, TContainer const & cont)
-{
-    SEQAN_CHECKPOINT;
-    return it == begin(cont, Standard());
-}
-
-template <typename T, typename TContainer>
-inline SEQAN_HOST_DEVICE bool
-atBegin(T const & it, TContainer & cont)
-{
-    SEQAN_CHECKPOINT;
-    return it == begin(cont, Standard());
-}
-
-template <typename T, typename TContainer>
-inline SEQAN_HOST_DEVICE bool
 atBegin(T & it, TContainer & cont)
```

Comment:
The non-const Version implies both a cinst interface and a non-const interface.

Having only a const interface on the other hand results in comparisons between const iterators and non const iterators (if we had an explicit `cbegin()` this could be used). Doing a cast might also work but makes the code less readable.